packages/ui-editable/src/Editable/props.ts

```diff
@@ -26,66 +26,102 @@ import PropTypes from 'prop-types'
 
 import type { PropValidators } from '@instructure/shared-types'
 
-type EditableOwnProps = {
+type containerProps = (props?: Record<string, any>) => {
+  ref: (el: Element | null) => void
+  onMouseOver: (event: React.MouseEvent) => void
+  onMouseOut: (event: React.MouseEvent) => void
+  onMouseDown: (event: React.MouseEvent) => void
+  onKeyUp: (event: React.KeyboardEvent) => void
+  readOnly?: boolean
+} & Record<string, any>
+
+type viewerProps = (props?: Record<string, any>) => {
   mode: 'view' | 'edit'
-  onChangeMode: (...args: any[]) => any
-  children?: (...args: any[]) => any
-  render?: (...args: any[]) => any
-  value?: any
-  onChange?: (...args: any[]) => any
   readOnly?: boolean
-  elementRef?: (element: Element | null) => void
-}
+} & Record<string, any>
 
-type PropKeys = keyof EditableOwnProps
+type editorProps = (props?: Record<string, any>) => {
+  mode: 'view' | 'edit'
+  onBlur: () => void
+  editorRef: (el: HTMLElement | null) => void
+  readOnly?: boolean
+} & Record<string, any>
 
-type AllowedPropKeys = Readonly<Array<PropKeys>>
+type editButtonProps = (props?: Record<string, any>) => {
+  onClick: () => void
+  onFocus: () => void
+  onBlur: () => void
+  isVisible: boolean
+  buttonRef: (el: Element) => void
+  readOnly?: boolean
+} & Record<string, any>
 
-type EditableProps = EditableOwnProps
+type renderProps = {
```

Comment:
Since this is exported from the component, we have to prefix it, so please rename it to `EditableRenderProps`.